pubsub/schemas/publish_proto_messages.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+// [START pubsub_publish_proto_messages]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/pubsub"
+	statepb "github.com/GoogleCloudPlatform/golang-samples/internal/pubsub/schemas"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+func publishProtoMessages(w io.Writer, projectID, topicID string) error {
+	// projectID := "my-project-id"
+	// topicID := "my-topic"
+	ctx := context.Background()
+	client, err := pubsub.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("pubsub.NewClient: %v", err)
+	}
+
+	state := &statepb.State{
+		Name:     "Alaska",
+		PostAbbr: "AK",
+	}
+
+	// Get the topic encoding type.
+	t := client.Topic(topicID)
+	cfg, err := t.Config(ctx)
+	if err != nil {
+		return fmt.Errorf("topic.Config err: %v", err)
+	}
+	encoding := cfg.SchemaSettings.Encoding
+
+	var msg []byte
+	switch encoding {
```

Comment:
Do you need to worry about ascii/textproto as well, or only binary and json?

Reply:
Only binary and Json are supported now, but others may get added later.